stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -302,6 +302,16 @@ function Base.showarg(io::IO, v::Transpose, toplevel)
     toplevel && print(io, " with eltype ", eltype(v))
     return nothing
 end
+function Base.show(io::IO, v::Adjoint{<:Real, <:AbstractVector})
```

Comment:
Note that this omits vectors of complex numbers... because I thought that printing `0 + 3im` when the element is in fact `0 - 3im` here is worse than failing to note the container type:
```julia
julia> (transpose([1,2,3im]), adjoint([1,2,3im]))
(transpose(Complex{Int64}[1 + 0im, 2 + 0im, 0 + 3im]), Complex{Int64}[1 + 0im 2 + 0im 0 - 3im])
```
It also doesn't change how e.g. an array of matrices is printed, for the same reason -- printing `[1 2; 3 4]` which isn't `==` the element seems confusing:
```julia
julia> ([[1 2; 3 4], [5 6; 7 8]]',)
(Adjoint{Int64, Matrix{Int64}}[[1 3; 2 4] [5 7; 6 8]],)
```

Reply:
I agree with the choices here

Reply:
Should we add tests for the complex case, making it explicit that this was a design choice and not an oversight?

Reply:
I think the restrictions to `v::Adjoint{<:Real, <:AbstractVector}` and  `v::Transpose{<:Number, <:AbstractVector}` are fairly obviously deliberate.

For the complex case, it would not be crazy to print `adjoint(conj([1 + 0im, 2 + 0im, 0 - 3im]))` as this also resolves to the right object (since `conj` is eager). Maybe I'd rather not have explicit tests blocking that.